mesh/mesh.hpp

```diff
@@ -1010,6 +1014,44 @@ class Mesh
    /// store these in the Mesh::attributes and Mesh::bdr_attributes arrays.
    virtual void SetAttributes();
 
+   /// @brief Prepares a marker array corresponding to an array of element
+   /// attributes
+   ///
+   /// @param max_attr Number of entries to create in the @a marker array
+   /// @param attrs    An array of attribute numbers which should be activated
+   /// @param marker   Output array indicating active/inactive attributes
+   ///
+   /// The marker array will be of size max_attr and it will contain only zeroes
+   /// and ones. Ones indicate which attribute numbers are present in the attrs
+   /// array. In the special case when attrs has a single entry equal to -1 the
+   /// marker array will contain all ones.
```

Comment:
I'm not sure this special case is desirable.

Reply:
It's something I use frequently. It is a convenient shortcut when specifying a long list of attributes would be a hassle. Why would it be undesirable?

Reply:
To me, it's non-obvious behavior. If I saw passing -1 as an attribute, I would first think that it's a bug or mistake in the logic, and only until reading the (last sentence of the) documentation would it make sense.

Isn't it equivalent to this:

```
mesh.AttrToMarker(mesh.attributes, markers);
```

and

```
mesh.BdrAttrToMarker(mesh.bdr_attributes, bdr_markers);
```

which to me is much more self-documenting (i.e., intent is clearly indicated in the code)?

Reply:
I look at this from the point of view of an application developer. This is a shortcut I like to use so it appears in many applications that I have written. This means that in each application I need to write some small piece of code to offer such an option to the user and then parse that option and translate it into a line like those that you have suggested. It would be easier for me if the library recognized this special case. Then the application can simply parse the user selected set of attributes from the command line and pass them directly to the library. If the user specifies '-1' the library knows what to do with this and no code needs to be written in any application to process this special case.

That being said you also make a good point from the library developer point of view. If you want me to remove this special case I will.

Reply:
Another relevant point is that this behavior has existed, for years, in the standalone function `AttrToMarker` located in the `miniapps/common/mesh_extras.?pp` files. This function. has just never been needed in the example codes until now.

Reply:
You're right, I wasn't looking at it from the perspective of passing user input to these functions, and I can see how that would be convenient. But my vote would still be to remove this special case — a helper function to translate a special input (e.g. "-1") to mean all attributes belongs better in application code rather than in the core library (different applications could have different conventions for how BCs are specified)

Reply:
Nothing here would prevent an application from using a different convention.

Reply:
If we had another function, e.g. `MarkAllAttr(int max_attr, Array<int> &marker)` that sets `marker` to all 1's of size `max_attr`, it would be clearer and avoid the `-1` encoding. I suppose this would be less convenient for the application, since it has to check for its special case and call a different function, e.g.

`if (attrs[0] == -1) MarkAllAttr(max_attr, marker); else AttrToMarker(max_attr, attrs, marker);`

Is that inconvenient enough to justify complicating the interface to this function? It seems like there would be some logic to set `attrs[0] = -1` in the application. If it is already setting it to `-1` in some special case, calling `MarkAllAttr` in that case wouldn't be a significant inconvenience, right? Or am I underestimating the inconvenience?

Reply:
The main convenience is seen on the command line. If a user wants to specify all attributes they don't need to explicitly list them on the command line. Once we're inside an application I'm not overly concerned with the logic. This is just a convenient way to avoid duplicating the same logic in different applications. @dylan-copeland, if you also think this is confusing, I can remove it and just add this function to the two examples that are added in this PR.

Reply:
If I understood correctly, the advantage is an application can have a command-line argument with an array of attributes, e.g. `-att 1 2 3 4 5`, or `-att -1` for all attributes, e.g. `miniapps/electromagnetics/maxwell.cpp`. Without supporting the `-1` case, the application would need an argument like `-allatt` and then have logic to call `MarkAllAttr` in the case that `-allatt` was used. That's not a serious inconvenience, but that logic would need to be repeated in every such application. I don't think it matters too much, and I'm fine with keeping it as is, if there is a reasonable justification like this.

Reply:
I think best way to handle `-1` in this case would be a one-liner in application code after reading the command-line arguments:
```cpp
if (attrs.Size() == 1 && attrs[0] == -1) { attrs = mesh.attributes; }
```
No need to introduce a special case to the library API in my opinion.

Reply:
In the weekly PR meeting we decided to remove these new helper functions from the mesh class